controller_manager: # Node name
  ros__parameters:
    update_rate: 30 # Has to be an integer otherwise errors are encountered

    diff_controller:
      type: diff_drive_controller/DiffDriveController

    joint_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    imu_broadcaster:
      type: imu_sensor_broadcaster/IMUSensorBroadcaster 


diff_controller:
  ros__parameters:
    
    publish_rate: 50.0 #60.0 # Has to be a double otherwise errors are encountered

    base_frame_id: base_footprint

    left_wheel_names: ['left_wheel_joint']
    right_wheel_names: ['right_wheel_joint']
    wheel_separation: 0.134
    wheel_radius: 0.0335

    pose_covariance_diagonal: [0.05, 0.05, 0.0, 0.0, 0.0, 0.01]
    twist_covariance_diagonal: [0.05, 0.05, 0.0, 0.0, 0.0, 0.01] 

    # pose_covariance_diagonal: [0.001, 0.001, 0.0, 0.0, 0.0, 0.01]
    # twist_covariance_diagonal: [0.001, 0.001, 0.0, 0.0, 0.0, 0.01] 

    # pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 0.01]
    # twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 0.01] 

    # pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    # twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0] 

    # Whether the input has a time stamp or not
    use_stamped_vel: false

imu_broadcaster:
  ros__parameters:

    sensor_name: mpu6050
    frame_id: imu_link

    static_covariance_orientation: [0.005, 0.0, 0.0, 0.0, 0.005, 0.0, 0.0, 0.0, 0.005] # Imu orientation not fused so this might not matter
    static_covariance_angular_velocity: [5.76e-04, 0.0, 0.0, 0.0, 5.76e-04, 0.0, 0.0, 0.0, 5.76e-04]
    static_covariance_linear_acceleration: [9.8067e-03, 0.0, 0.0, 0.0, 9.8067e-03, 0.0, 0.0, 0.0, 9.8067e-03]

    # static_covariance_orientation: [0.005, 0.0, 0.0, 0.005, 0.0, 0.0, 0.005, 0.0, 0.0] # Imu orientation not fused so this might not matter
    # static_covariance_angular_velocity: [5.76e-04, 0.0, 0.0, 5.76e-04, 0.0, 0.0, 5.76e-04, 0.0, 0.0]
    # # static_covariance_linear_acceleration: [9.8067e-03, 0.0, 0.0, 9.8067e-03, 0.0, 0.0, 0.0, 0.0, 0.0]
    # static_covariance_linear_acceleration: [9.8067e-03, 0.0, 0.0, 9.8067e-03, 0.0, 0.0, 9.8067e-03, 0.0, 0.0]